state-chain/custom-rpc/src/lib.rs

```diff
@@ -1277,6 +1280,22 @@ where
 			.cf_witness_count(self.unwrap_or_best(at), pallet_cf_witnesser::CallHash(hash.into()))
 			.map_err(to_rpc_error)
 	}
+
+	fn cf_decode_runtime_events(&self, events: Vec<String>) -> RpcResult<Vec<String>> {
```

Comment:
I don't think we want to have to provide the events as an argument. 

We want to call this a bit like `get_events(block_hash: Hash) -> Vec<serde_json::Value>` and then:
1. Use the runtime api to read the events storage as a Vec<OpaqueEvent>, where OpaqueEvent is just the raw scale-encoded Vec<u8> representing each RuntimeEvent.
2. This method will get the correct metadata (at the provided hash) and then decode each Vec<u8> as a `scale_value::Value<RuntimeEvent::TYPE_ID>` using scale_value::decode_as_type.
3. The scale_value is then re-encoded to a serde_json::Value (see other comment).

If the transcoding of any OpaqueEvent fails, we log an error and skip it.